src/engine/MILPEncoder.cpp

```diff
@@ -187,3 +203,152 @@ void MILPEncoder::encodeMaxConstraint( GurobiWrapper &gurobi, MaxConstraint *max
+        terms.clear(); 
+
+        // upper bound of x_b
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", sourceVariable ) ) );
+        gurobi.addLeqIndicatorConstraint( binVarName, binVal, terms, sourceUb );  
+        terms.clear();
+
+        // lower bound of x_f
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", targetVariable ) ) );
+        gurobi.addGeqIndicatorConstraint( binVarName, binVal, terms, y_l );  
+        terms.clear(); 
+
+        // upper bound of x_f
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", targetVariable ) ) );
+        gurobi.addLeqIndicatorConstraint( binVarName, binVal, terms, y_u );  
+        terms.clear();       
+
+        //
+        // Constraints where x_b <= 0
+        // Upper line is secant and lower line is tangent for an overapproximation with a linearization.
+        //
+
+        binVal = 0;
+
+        // tangent line: x_f = tangentSlope * (x_b - tangentPoint) + yAtTangentPoint
+        tangentPoint = ( sourceLb + 0 ) / 2;
+        yAtTangentPoint = sigmoid->sigmoid( tangentPoint );
+        tangentSlope = sigmoid->sigmoidDerivative( tangentPoint );
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", targetVariable ) ) );
+        terms.append( GurobiWrapper::Term( -tangentSlope, Stringf( "x%u", sourceVariable ) ) );
+        gurobi.addGeqIndicatorConstraint( binVarName, binVal, terms, -tangentSlope * tangentPoint + yAtTangentPoint );
+        terms.clear();
+
+        // secant line: x_f = secantSlope * (x_b - sourceLb) + y_l
+        y_u = sigmoid->sigmoid( 0 );
+        y_l = sigmoid->sigmoid( sourceLb );
+        secantSlope = ( y_u - y_l ) / ( 0 - sourceLb );
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", targetVariable ) ) );
+        terms.append( GurobiWrapper::Term( -secantSlope, Stringf( "x%u", sourceVariable ) ) );
+        gurobi.addLeqIndicatorConstraint( binVarName, binVal, terms, -secantSlope * sourceLb + y_l );
+        terms.clear();
+
+        // lower bound of x_b
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", sourceVariable ) ) );
+        gurobi.addGeqIndicatorConstraint( binVarName, binVal, terms, sourceLb );  
+        terms.clear(); 
+
+        // upper bound of x_b
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", sourceVariable ) ) );
+        gurobi.addLeqIndicatorConstraint( binVarName, binVal, terms, 0 );  
+        terms.clear();
+
+        // lower bound of x_f
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", targetVariable ) ) );
+        gurobi.addGeqIndicatorConstraint( binVarName, binVal, terms, y_l );  
+        terms.clear();
+
+        // upper bound of x_f
+        terms.append( GurobiWrapper::Term( 1, Stringf( "x%u", targetVariable ) ) );
+        gurobi.addLeqIndicatorConstraint( binVarName, binVal, terms, y_u );  
+        terms.clear(); 
+
+        _binVarIndex++;
+        return;
+    }
+
```

Comment:
add "else {" here with respect to "if ( sourceLb < 0 && sourceUb > 0 )", for the sake of code readability.